src/handlers/swap_handlers.ts

```diff
@@ -30,7 +30,55 @@ export class SwapHandlers {
         this._swapService = swapService;
     }
     public async getSwapQuoteAsync(req: express.Request, res: express.Response): Promise<void> {
-        // parse query params
+        res.status(HttpStatus.OK).send(
+            await this._calculateSwapQuoteAsync(parseGetSwapQuoteRequestParams(req), req.header('0x-api-key')),
+        );
+    }
+    // tslint:disable-next-line:prefer-function-over-method
+    public async getSwapTokensAsync(_req: express.Request, res: express.Response): Promise<void> {
+        const tokens = TokenMetadatasForChains.map(tm => ({
+            symbol: tm.symbol,
+            address: tm.tokenAddresses[CHAIN_ID],
+            name: tm.name,
+            decimals: tm.decimals,
+        }));
+        const filteredTokens = tokens.filter(t => t.address !== NULL_ADDRESS);
+        res.status(HttpStatus.OK).send({ records: filteredTokens });
+    }
+    // tslint:disable-next-line:prefer-function-over-method
+    public async getSwapPriceAsync(req: express.Request, res: express.Response): Promise<void> {
+        const params = parseGetSwapQuoteRequestParams(req);
+        if (params.rfqt === undefined) {
+            params.rfqt = {
+                intentOnFilling: false,
+            };
+        }
+        params.skipValidation = true;
+        const quote = await this._calculateSwapQuoteAsync(params, req.header('0x-api-key'));
+        const { price, value, gasPrice, gas, protocolFee, buyAmount, sellAmount } = quote;
+        res.status(HttpStatus.OK).send({ price, value, gasPrice, gas, protocolFee, buyAmount, sellAmount });
+    }
+    // tslint:disable-next-line:prefer-function-over-method
+    public async getTokenPricesAsync(req: express.Request, res: express.Response): Promise<void> {
+        const symbolOrAddress = req.query.sellToken || 'WETH';
+        const baseAsset = getTokenMetadataIfExists(symbolOrAddress, CHAIN_ID);
+        if (!baseAsset) {
+            throw new ValidationError([
+                {
+                    field: 'sellToken',
+                    code: ValidationErrorCodes.ValueOutOfRange,
+                    reason: `Could not find token ${symbolOrAddress}`,
+                },
+            ]);
+        }
+        const unitAmount = new BigNumber(1);
+        const records = await this._swapService.getTokenPricesAsync(baseAsset, unitAmount);
+        res.status(HttpStatus.OK).send({ records });
+    }
+    private async _calculateSwapQuoteAsync(
+        params: GetSwapQuoteRequestParams,
```

Comment:
feel like apiKey should live in `GetSwapQuoteRequestParams`. Don't think it needs to travel separately just because it originates via a Header.

Reply:
Addressed in https://github.com/0xProject/0x-api/pull/172/commits/1b9b20c212f7d5c0bf698f77e04a95dc6c93ec35